### PR TITLE
BI 8761 dynamic order confirmation page

### PR DESCRIPTION
**Pull request by phudson-ch**
- Add LP & LLP integration tests for order.confirmation.controller
- Add director & secretary checks to LTD company integration test

### CODE REVIEW

File: src/controllers/basket.controller.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const render = async (req: Request, res: Response, next: NextFunction) =>
         const checkoutApiResponse: ApiResponse<Checkout> = await checkoutBasket(accessToken);
 
         const checkoutId = checkoutApiResponse.resource?.reference;
-        logger.info(`Checkout created, checkout_id=${checkoutId}, user_id=${userId}`);
+        logger.info(`XXXXX Checkout created, checkout_id=${checkoutId}, user_id=${userId}`);
```
Comment (kinpang-CH) [inline src/controllers/basket.controller.ts:28]: Removed unneeded characters

File: src/controllers/order.confirmation.controller.ts [hunk 2]
```diff
@@ -46,12 +46,14 @@ export const render = async (req: Request, res: Response, next: NextFunction) =>
 
         logger.info(`Order retrieved order_id=${order.reference}, user_id=${userId}`);
 
+        const item = order.items[0];
+
         const orderDetails = {
             referenceNumber: order.reference,
-            referenceNumberAriaLabel: order.reference.replace(/-/g, " hyphen ")
+            referenceNumberAriaLabel: order.reference.replace(/-/g, " hyphen "),
+            itemOptions: item.itemOptions
```
Comment (kinpang-CH) [inline src/controllers/order.confirmation.controller.ts:54]: Check to see if change can be reverted

File: src/test/controllers/order.confirmation.controller.integration.test.ts [hunk 3]
```diff
@@ -23,16 +28,16 @@ const ITEM_KINDS = [{
...
+    {
+        kind: "item#certified-copy",
+        name: "certified-copy",
+        url: "/orderable/certified-copies/CCD-123456-123456"
+    },
+    {
+        kind: "item#missing-image-delivery",
+        name: "missing-image-delivery",
+        url: "/orderable/missing-image-delivery/MID-123456-123456"
+    }];
```
Comment (kinpang-CH) [inline src/test/controllers/order.confirmation.controller.integration.test.ts:40]: Revert indentation for 31-40

File: src/test/controllers/order.confirmation.controller.integration.test.ts [hunk 4]
```diff
@@ -48,15 +53,140 @@ describe("order.confirmation.controller.integration", () => {
...
+
+        getOrderStub = sandbox.stub(apiClient, "getOrder").returns(Promise.resolve(certificateOrderResponse));
 
         chai.request(testApp)
             .get(`/orders/${ORDER_ID}/confirmation?ref=orderable_item_${ORDER_ID}&state=1234&status=paid&itemType=certificate`)
             .set("Cookie", [`__SID=${SIGNED_IN_COOKIE}`])
             .end((err, resp) => {
                 if (err) return done(err);
-                const $ = cheerio.load(resp.text);
+                const $ = cheerio.load(resp.text, {decodeEntities: false});
```
Comment (kinpang-CH) [inline src/test/controllers/order.confirmation.controller.integration.test.ts:189]: Remove {decodeEntities: false}

File: src/service/LLPCertificateItemMapper.ts [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                        {
+                            visuallyHiddenText: "current members names"
+                        }
+                    ]
+                }
+            }
+        ];
+    }
+
+    static mapRegisteredOfficeAddress = (itemOptions: CertificateItemOptions): string => {
```
Comment (kinpang-CH) [inline src/service/LLPCertificateItemMapper.ts:98]: Can belong into its own class, as it's re-used several times

**Comment by kinpang-CH** [comment]
Remove static methods usage.

**Review by cgregoryCH — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Pull request merged**